src/bsoncxx/lib/bsoncxx/v1/element/view.cpp

```diff
@@ -12,19 +12,337 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include <bsoncxx/v1/element/view.hpp>
+#include <bsoncxx/v1/element/view.hh>
 
 //
 
+#include <bsoncxx/v1/document/view.hpp>
+#include <bsoncxx/v1/exception.hpp>
+
+#include <bsoncxx/v1/types/value.hh>
+#include <bsoncxx/v1/types/view.hh>
+
+#include <string>
+
+#include <bsoncxx/private/bson.hh>
+#include <bsoncxx/private/immortal.hh>
+#include <bsoncxx/private/make_unique.hh>
 #include <bsoncxx/private/type_traits.hh>
 
 namespace bsoncxx {
 namespace v1 {
 namespace element {
 
+using code = v1::element::view::errc;
+
 static_assert(is_semiregular<view>::value, "bsoncxx::v1::element::view must be semiregular");
 static_assert(is_nothrow_moveable<view>::value, "bsoncxx::v1::element::view must be nothrow moveable");
 
+class alignas(BSONCXX_PRIVATE_MAX_ALIGN_T) view::impl {
+   private:
+    enum : std::size_t {
+        _padding_size = sizeof(view::_storage)       // Total reserved.
+                        - sizeof(void*)              // _raw
+                        - 3u * sizeof(std::uint32_t) // _length, _offset, and _keylen.
+                        - 1u                         // _is_valid (final byte).
+                        - 0u,
+    };
+
+    // `_padding_size == 3` given `sizeof(void*) == 8`.
+    static_assert(_padding_size < sizeof(view::_storage), "sizeof(impl) must not exceed reserved storage size");
```

Comment:
We do not support any platforms where `sizeof(void*)` or `alignof(void*)` exceed `8`, although they may equal `4` when targeting certain 32-bit platforms.